### PR TITLE
Refactored the parser and analyzer as well as packages

### PR DESCRIPTION
**Pull request by edwardhuahan**
I removed a lot of boolean logic and implemented some classes for containing the data, this was because the boolean logic could've been much more simplified to be readable

I tested everything within Postman using the data_small.json file we're using and the output was the exact same, feel free to test it yourself as well.

Going forward I still want to add more Classes for holding data instead of using a huge nested Arraylist but I still need to work on that

I also changed the package structure to be much more concise

image

image

### CODE REVIEW

**Review by ZZBP — APPROVED** [review]
Hi Edward! Amazing work refactoring the parser! Conversation and DialogueList certainly help make parseTranscript more concise. Though these two classes are pretty self-explanatory, some documentation would be great!

File: backend/src/main/java/com/speakfluid/backend/entities/Conversation.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.speakfluid.backend.entities;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class Conversation {
+
```
Comment (Lemiex) [inline backend/src/main/java/com/speakfluid/backend/entities/Conversation.java]: add documentation to let everyone understand the great work u did :D

**Review by Lemiex — APPROVED** [review]
Great idea making Conversation and DialogueLists! Gets rid of some ugly <<<>>> type nesting! Just add documentation in all ur files are you're solid

**Review by nm-le — APPROVED** [review]
Thank you for cleaning up the code :) Would we want to add a Transcript class as well?

File: backend/src/main/java/com/speakfluid/backend/usecases/WozTranscriptAnalysisInteractor.java [hunk 2]
```diff
@@ -1,6 +1,12 @@
 package com.speakfluid.backend.usecases;
 
 import com.speakfluid.backend.entities.*;
+import com.speakfluid.backend.entities.message.Transcript;
```
Comment (Lemiex) [inline backend/src/main/java/com/speakfluid/backend/usecases/WozTranscriptAnalysisInteractor.java]: why are there import statements line 4-9 if we already imported all the entities in line 3?

File: backend/src/main/java/com/speakfluid/backend/usecases/TranscriptParser.java [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+                dialogueList.add(new Dialogue<WozMessage>(chatbotMessage, userMessage));
+                chatbotMessage = new ArrayList<>();
+                userMessage = new ArrayList<>();
+            }
+
+        }
+        return dialogueList;
+    }
+
+//     public static void main(String[] args) {
```
Comment (Lemiex) [inline backend/src/main/java/com/speakfluid/backend/usecases/TranscriptParser.java]: what's this and why is it commented out?

**Review by Lemiex — APPROVED** [review]
Loving the new directory structure

File: backend/src/main/java/com/speakfluid/backend/usecases/WozTranscriptAnalysisInteractor.java (hunk 2 above)
Comment (edwardhuahan) [reply backend/src/main/java/com/speakfluid/backend/usecases/WozTranscriptAnalysisInteractor.java]: Oh lol that's generated from intelliJ

File: backend/src/main/java/com/speakfluid/backend/usecases/TranscriptParser.java (hunk 3 above)
Comment (edwardhuahan) [reply backend/src/main/java/com/speakfluid/backend/usecases/TranscriptParser.java]: LOL I got that from Minh

Comment (Lemiex) [reply backend/src/main/java/com/speakfluid/backend/usecases/TranscriptParser.java]: @nm-le help

**Review by ZZBP — APPROVED** [review]
Amazing work refactoring all the packages!

**Review by nm-le — APPROVED** [review]

**Pull request merged**